compiler/pragmas.nim

```diff
@@ -771,6 +771,8 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: int,
       of wNoreturn:
         noVal(it)
         incl(sym.flags, sfNoReturn)
+        if sym.ast[paramsPos][0].kind != nkEmpty:
+          localError(sym.ast[paramsPos][0].info, errNoReturnWithReturnTypeNotAllowed)
```

Comment:
I'll merge this now anyway, but this check is not 100% correct, ``void`` as a return type would be valid.